### PR TITLE
Datatables filter

### PR DESCRIPTION
**Pull request by marc-hanheide**
column-based filter and option to show all matching entries

### CODE REVIEW

**Comment by marc-hanheide** [comment]
@pet1330 if you are curious how I did this ;-)

File: resources/assets/js/datatables.js [hunk 1]
```diff
@@ -11,7 +31,7 @@ $(document).ready(function() {
         // stateSave: true,
         bProcessing: true,
         deferRender: false,
-        lengthMenu: [[10, 20, 30, 40, 50, 100, 200, 500], [10, 20, 30, 40, 50, 100, 200, 500]],
+        lengthMenu: [[10, 20, 30, 40, 50, 100, 200, 500, -1], [10, 20, 30, 40, 50, 100, 200, 500, "All"]],
```
Comment (pet1330) [inline resources/assets/js/datatables.js]: I would advise against allowing the user to disable pagination. This has the potential to put a massive strain on the database server, but more importantly, can make the user's page very slow and unresponsive as they try to render huge datasets.

**Comment by pet1330** [comment]
Other than the above two comments, it looks good! :smiley:

**Comment by marc-hanheide** [comment]
Yes, but as long as the reports (pdf/xlsx) exports are bound to the entries currently displayed in the data table, the semantics of "export all that match the query" needs to be facilitated by the use cases the user present. I expect this to be used in conjunction with column-based filter to implement the typical uses that have been put to me, e.g., show all enrolled students of CompSci, or all overseas students etc. If not in the data tables than we'd have to detach the export from the pagination. This is the quick way of addressing the use cases.

**Pull request merged**